modules/gapi/include/opencv2/gapi/s11n.hpp

```diff
@@ -142,7 +154,105 @@ namespace I {
+    #endif // !defined(GAPI_STANDALONE)
+
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, const cv::gapi::wip::IStreamSource::Ptr &);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is,       cv::gapi::wip::IStreamSource::Ptr &);
+
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, const cv::detail::VectorRef &);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is,       cv::detail::VectorRef &);
+
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, const cv::detail::OpaqueRef &);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is,       cv::detail::OpaqueRef &);
+
+    // FIXME: moved here due to clang lookup issue (see the comment above)
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, const cv::gimpl::RcDesc &rc);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is,       cv::gimpl::RcDesc &rc);
+
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, const cv::gimpl::Op &op);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is,       cv::gimpl::Op &op);
+
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, const cv::gimpl::Data &op);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is,       cv::gimpl::Data &op);
+
+
+    // Generic: variant serialization //////////////////////////////////////////////
+    namespace detail {
+    template<typename V>
+    I::OStream& put_v(I::OStream&, const V&, std::size_t) {
+        GAPI_Assert(false && "variant>>: requested index is invalid");
+    };
+    template<typename V, typename X, typename... Xs>
+    I::OStream& put_v(I::OStream& os, const V& v, std::size_t x) {
+        return (x == 0u)
+            ? os << cv::util::get<X>(v)
+            : put_v<V, Xs...>(os, v, x-1);
+    }
+    template<typename V>
+    I::IStream& get_v(I::IStream&, V&, std::size_t, std::size_t) {
+        GAPI_Assert(false && "variant<<: requested index is invalid");
+    }
+    template<typename V, typename X, typename... Xs>
+    I::IStream& get_v(I::IStream& is, V& v, std::size_t i, std::size_t gi) {
+        if (i == gi) {
+            X x{};
+            is >> x;
+            v = std::move(x);
+            return is;
+        } else return get_v<V, Xs...>(is, v, i+1, gi);
+    }
+    } // namespace detail
+
+    template<typename... Ts>
+    I::OStream& operator<< (I::OStream& os, const cv::util::variant<Ts...> &v) {
+        os << (uint32_t)v.index();
+        return detail::put_v<cv::util::variant<Ts...>, Ts...>(os, v, v.index());
+    }
+    template<typename... Ts>
+    I::IStream& operator>> (I::IStream& is, cv::util::variant<Ts...> &v) {
+        int idx = -1;
+        is >> idx;
+        GAPI_Assert(idx >= 0 && idx < (int)sizeof...(Ts));
+        return detail::get_v<cv::util::variant<Ts...>, Ts...>(is, v, 0u, idx);
+    }
+
+    // Generic: stl structures serialization //////////////////////////////////////////////
     template<typename K, typename V>
     GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, const std::map<K, V> &m) {
```

Comment:
Please order `<<` and `>>` for different data structures properly. `<<(map)` should be followed by `>>(map)`.

Reply:
Ordered